proof_of_leadership/risc0/proof_statements/src/proof_of_leadership.rs

```diff
@@ -0,0 +1,95 @@
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+use crypto_bigint::{U256, Encoding, CheckedMul, CheckedSub};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+pub struct LeaderPublic {
+    pub cm_root: [u8; 32],
+    pub epoch_nonce: [u8; 32],
+    pub slot: u64,
+    pub scaled_phi_approx: (U256, U256),
+    pub nullifier: cl::Nullifier,
+    pub updated_commitment: cl::NoteCommitment,
```

Comment:
mmm, this is not known to the verifier I think?

Reply:
They don't need to know it, it'll be provided by the winning leader, the verifier just checks that this note commitment was evolved from the winning note (coin).

This updated commitment gets added to the "leader" commitment set

Reply:
sorry I was confused, all good